Backend/Backend/Controllers/ProfileController.cs

```diff
@@ -82,14 +82,21 @@ public async Task<ActionResult> SetupProfile([FromForm] ProfileSetupRequest setu
     {
         try
         {
-            this.LogDebugControllerAPICall(_logger, callerName: nameof(SetupProfile));
+            this.LogDebugControllerAPICall(_logger);
 
             // Identify User from JWT Token
             User? user = await _authService.IdentifyUserAsync(HttpContext);
 
             // If User is not found, return 404
             if (user == null)
                 return NotFound("User does not exist.");
+            
+            // Validate file extension
+            if (setupRequest.Avatar is not null) {
+                string extension = Path.GetExtension(setupRequest.Avatar.FileName);
+                if (extension != ".jpg" && extension != ".png" && extension != ".jpeg" && extension != ".gif")
```

Comment:
Should probably use an enum here instead, and relegate that check to saveUserAvatar